packages/discord.js/typings/index.d.ts

```diff
@@ -3658,7 +3658,7 @@ export interface TextBasedChannelFields extends PartialTextBasedChannelFields {
   bulkDelete(
     messages: Collection<Snowflake, Message> | readonly MessageResolvable[] | number,
     filterOld?: boolean,
-  ): Promise<Collection<Snowflake, Message>>;
+  ): Promise<Collection<Snowflake, Message | undefined>>;
```

Comment:
`PartialMessage` should be added here.

If we consider the case that a array of uncached message ids are passed to this method but we also specify the message partial, we will get back a `Message` structure but it will be partial.